API.md

```diff
@@ -0,0 +1,203 @@
+# CodeReady Analytics Core API Specifications
+
+CodeReady Analytics is an open-source analytics platform aimed primarily at assisting developers in getting insights and recommendations on the selected dependency/package stack used for developing their applications.
+
+This documentation is for developers who are interested in integrating their services or applications with CodeReady Analytics platform.
+
+## **Supported ecosystems**
+
+Below are the ecosystems that CodeReady Analytics currently supports.  We will be adding support for more ecosystems in the near future.
+| | |
+|-|-|
+|**Ecosystem**|**Manifest file**|
+| Java Maven | pom.xml |
+| Python Pypi | requirements.txt |
+| NodeJS | package.json |
+
+## **Prerequisites**
+
+Any new integration with the CodeReady Analytics platform will have to do that using an existing 3-scale API management gateway setup. Please reach out to us by raising an issue [here](https://github.com/fabric8-analytics/fabric8-analytics-vscode-extension/issues)
+
+## **Initiate Dependency Analysis**
+
+Submits a dependency stack analysis request
+
+- **URL**
+  `https://<3scale-api-gateway-url>/api/v1/stack-analyses`
+
+- **Method**
+  `POST`
+
+- **Response Format**
+  `jSON`
+
+- **Path Parameters**
+  `None`
+
+- **Query Parameters**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | user_key | required | 3-scale API management authorization user key. | 421249d63 |
+
+- **Form Data**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | manifest[] | required | Ecosystem specific Manifest file(s) available in the application. One can provide multiple manifest files to invoke the method. | @pom.xml |
+  | filePath[] | required | Corresponding manifest file’s directory information. Care has to be taken to map the manifest filenames and file paths correctly. | /home/JohnDoe |
+
+- **Headers**
+  | | | | |
+  |-|-|-|-|
+  |**Name**|**Required**|**Description**|**Example**|
+  | source | required | Source of the request. | "vscode" |
+  | ecosystem | required | Ecosystem | maven/pypi/npm |
+
+- **Sample Request with cURL**
+
+  ``` 
+  $ curl -k -F "manifest[]=@./pom.xml" -F "filePath[]=/home/JohnDoe" https://https://api-244552094075730.prod.gw.apicast.io:443/api/v1/stack-analyses/?user_key=421249d63
+  ```
+
+- **Sample Responses**
+
+  **_Success_**
+
+    ```
+        {
+        "id": "421249d9e1e5464cbf3e77dde4941463",
+        "status": "success",
+        "submitted_at": "2017-10-20 05:09:01.165068"
+        }
+    ```
+
+  **_Failure_**
+    ```
+        400:
+            {
+                "error": "Bad request"
+            }
+    ```
+
+    ```
+        401:
+            {
+                "error": "Authentication failed - could not decode JWT token"
+            }
+    ```
+
+## **Fetch Dependency Analysis**
+
+Fetches dependency stack analysis response using the request identifier returned by initiate dependency analysis.
```

Comment:
Fetches the dependency stack analysis response using the request identifier returned by the initiated dependency analysis.